src/journeys_utils.js

```diff
@@ -625,8 +612,10 @@ journeys_utils.finalHookups = function(templateId, audienceRuleId, storage, cta,
 			journeys_utils.animateBannerExit(banner);
 		})
 	})
-	journeys_utils._setupDismissBehavior('.branch-banner-continue', 'didClickJourneyContinue', storage, banner, templateId, audienceRuleId, metadata, testModeEnabled, branch_view);
-	journeys_utils._setupDismissBehavior('.branch-banner-close', 'didClickJourneyClose', storage, banner, templateId, audienceRuleId, metadata, testModeEnabled, branch_view);
+	journeys_utils._setupDismissBehavior('.branch-banner-continue', 'didClickJourneyContinue', storage, banner, templateId, audienceRuleId, metadata, testModeEnabled, branch_view, 'click');
+	journeys_utils._setupDismissBehavior('.branch-banner-close', 'didClickJourneyClose', storage, banner, templateId, audienceRuleId, metadata, testModeEnabled, branch_view, 'click');
+	journeys_utils._setupDismissBehavior('.branch-banner-dismiss-background', 'didClickJourneyBackgroundDismiss', storage, banner, templateId, audienceRuleId, metadata, testModeEnabled, branch_view, 'click'); // TODO add new dismissal_source
+	journeys_utils._setupDismissBehavior('.branch-banner-dismiss-background', 'didScrollJourneyBackgroundDismiss', storage, banner, templateId, audienceRuleId, metadata, testModeEnabled, branch_view, 'touchmove'); // TODO add new dismissal_source
```

Comment:
Should the TODO comments be removed and move into a Jira ticket?

Reply:
@yli-branch : I think if click, touchmove are both attached to the same element(branch-banner-dismiss-background) then on a touch device - the order will be touch event handler ( touch move)  first followed by mouse events ( click). Do you need to trigger the same event handler function ( _setupDismissBehavior)  twice for a touch device tap action ( once for touchmove and other time for click) ?

- You can use preventDefault in  event handler so that subsequent click/mouse event can be prevented from calling.
- Other way is to  check  if its a touchscreen device or not and invoke one of the handlers but this can fail on edge cases like touch screen laptops... so not a good way 
if(('ontouchmove' in window){
..
}
else
{
}

Reply:
@JagadeeshKaricherla-branch The product requirement actually has me targeting two separate use cases on the same touch device. When I tested this code on a mobile device simulation on chrome browser, only one call to _handleJourneyDismiss is made per interaction, and it was able to distinguish between a touchmove (scroll) and a click. Should I expect it to be different on a real device?